x-pack/plugins/reporting/public/management/components/report_info_flyout_content.tsx

```diff
@@ -50,6 +51,8 @@ export const ReportInfoFlyoutContent: FunctionComponent<Props> = ({ info }) => {
       ? moment.tz.guess()
       : uiSettings.get('dateFormat:tz');
 
+  const showKibanaVersion = Boolean(info.version) && !isServerless;
```

Comment:
3. Let's use the `config` object to determine this flag
```suggestion
  const showKibanaVersion = Boolean(info.version) && config.statefulSettings.enabled;
```